plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -1444,4 +1444,16 @@ public record RewrittenExpression(
             assignments = ImmutableList.copyOf(requireNonNull(assignments, "assignments is null"));
         }
     }
+
+    public static List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcClient jdbcClient, JdbcTableHandle tableHandle)
```

Comment:
It can be defined in `JdbcClient`. Also it could be deprecated to motivate people using the method you added.

Also try to not to use it, I mean let's port as much as reasonable in this PR to the method.

Reply:
I am thinking how should think about the `private final Optional<List<JdbcColumnHandle>> columns;` that in he `JdbcTableHandle`. It's seems a cached columns or adaption for the table handles that are not the `NamedRelation`. Therefore, if we (decide to) remove the static method in future(in a long term), the caller needs to know the handle type exactly, and the logic is still the same as the helper method, (1. check cached columns 2. get by new api), we need to inline the codes multi places if remove this api.

If we decide  to keep the helper method, then maybe shouldn't remove the method at all, just add a new api for convenient?

Reply:
Put the method into `JdbcMetadata`